arch/riscv/Kconfig

```diff
@@ -119,6 +119,11 @@ config RISCV_HAS_CPU_IDLE
 	help
 	  Does SOC has CPU IDLE instruction
 
+config RISCV_HAS_IRQ_PRIORITY
+	bool "Does SOC has IRQ priority setting"
+	help
+	  Does SOC has IRQ Priority setting
+
```

Comment:
I'm fine with adding this but adding `weak` symbols and blurring the distinction between PLIC and CLIC is something that must be justified a bit better. Why do you need this? To save some code or do you actually need this to be able to implement support for this board?

Reply:
I think it make improvement to introduce the weak symbol, but that is not mainly intention in this PR.
Rewrite it based on current code.

I consider to create another PR to introduce the weak symbol.